package NIndexAnn;

option optimize_for = SPEED;

message T4DArrayElement {
    optional uint64 ElementId = 1; // break id
    optional uint64 EntryKey = 2; // region id
    optional uint64 ItemKey = 3; // stream id
    optional bytes Data = 4; // stream value
}

message T4DArrayRow {
    repeated T4DArrayElement Elements = 1;
}

message TPortionPB {
    optional bytes Inv = 1;
    optional T4DArrayRow Row = 2;
    optional bytes SentenceLens = 3;
    optional bytes Arc = 4;
}
